### PR TITLE
feat(design-system): FloatingDrawer component

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**
No floating drawer in DS

**What is the chosen solution to this problem?**
Create it

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
+@use '~@talend/design-tokens/lib/tokens';
+
+.primitive-drawer {
+	width: 30rem;
```
Comment (Benrajalu) [inline packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss:4]:
```suggestion
	width: tokens.$coral-sizing-maximal;
```

File: packages/design-system/src/pages/ docs/Pages.stories.tsx [hunk 2]
```diff
@@ -107,8 +107,10 @@ const ItemWithDetails = ({
 						Item {itemId + 1}
 					</ButtonPrimary>
 				}
-				heading={<h3>Item {itemId + 1}</h3>}
-				footer={<ButtonSecondary onClick={() => setVisible(false)}>Close</ButtonSecondary>}
+				header={<h3>Item {itemId + 1}</h3>}
+				footer={(dialog: any) => (
```
Comment (gjou-tlnd) [inline packages/design-system/src/pages/ docs/Pages.stories.tsx:111]: that's a new component, let's type it strongly :)

**Review by gjou-tlnd — CHANGES_REQUESTED** [review]

**Comment by gjou-tlnd** [comment]
Chromatic doesn't seem to like you change:
https://www.chromatic.com/build?appId=62022f03c6654e003a637b65&number=465
Please write a play funciton opening the drawer in one of your stories so we have meaningful NR tests

File: packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+.primitive-drawer {
+	width: tokens.$coral-sizing-maximal;
+	height: 100%;
+	background: tokens.$coral-color-neutral-background;
+	display: flex;
+
+	&__header {
+		flex-grow: 0;
+		flex-shrink: 0;
+		flex-basis: 5.5rem;
```
[the remark this answers is not in the document]
Comment (jsomsanith-tlnd) [reply packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss:12]: Maxime told me go for 40px, I change it to `$coral-sizing-l`

File: packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import theme from './FloatingDrawer.scss';
+
+export type DrawerProps = {
+	disclosure?: React.ReactElement;
+	header?: React.ReactNode;
+	children: React.ReactNode;
+	footer?: React.ReactNode;
+	visible?: boolean;
+	onClose?: () => void;
+};
```
Comment (Benrajalu) [inline packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx:14]:
```suggestion
type WithDisclosure = {
   	disclosure: React.ReactElement;
   	visible?: never;
};

type Controlled = {
   	disclosure?: never;
   	visible: boolean;
};


export type DrawerProps = {
	header?: React.ReactNode;
	children: React.ReactNode;
	footer?: React.ReactNode;
	onClose?: () => void;
} & (WithDisclosure | Controlled);
```

Current typings open the component to impossible states

File: packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef } from 'react';
+import { Dialog, DialogDisclosure, useDialogState } from 'reakit';
+import PrimitiveDrawer from '../../Primitive/PrimitiveDrawer';
+
+import theme from './FloatingDrawer.scss';
+
+export type DrawerProps = {
+	disclosure?: React.ReactElement;
+	header?: React.ReactNode;
```
Comment (gjou-tlnd) [inline packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx:9]:
```suggestion
	header?: (dialog: DialogStateReturn) => React.ReactNode | React.ReactNode;
	footer?: (dialog: DialogStateReturn) => React.ReactNode | React.ReactNode;
```

Comment (jsomsanith-tlnd) [reply packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx:9]: Oh thanks !

File: packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
+@use '~@talend/design-tokens/lib/tokens';
+
+.floating-drawer {
+	position: absolute;
+	top: 0;
+	right: 0;
+	bottom: 0;
+	box-shadow: tokens.$coral-elevation-shadow-neutral-m;
+	z-index: tokens.$coral-elevation-layer-overlay;
+	transition: tokens.$coral-transition-fast;
```
Comment (Benrajalu) [inline packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.scss:10]:
```suggestion
	transition: transform tokens.$coral-transition-fast;
```

File: packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+		flex-basis: 0;
+		overflow: auto;
+
+		display: flex;
+	}
+
+	&__footer {
+		flex-grow: 0;
+		flex-shrink: 0;
+		flex-basis: 5.5rem;
```
Comment (Benrajalu) [inline packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss:30]: But not there?

Comment (jsomsanith-tlnd) [reply packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss:30]: UX opinion: the buttons' height is 36px, and having a 40px footer would be a little tight

File: packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx (hunk 4 above)
Comment (jsomsanith-tlnd) [reply packages/design-system/src/components/Drawer/variants/FloatingDrawer/FloatingDrawer.tsx]: Done

File: packages/design-system/src/pages/ docs/Pages.stories.tsx (hunk 2 above)
Comment (jsomsanith-tlnd) [reply packages/design-system/src/pages/ docs/Pages.stories.tsx]: Done

File: packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss (hunk 7 above)
Comment (Benrajalu) [reply packages/design-system/src/components/Drawer/Primitive/PrimitiveDrawer.scss:30]: Could be 40 + padding? Buttons would automatically make that zone taller

**Review by Benrajalu — APPROVED** [review]

**Review by gjou-tlnd — APPROVED** [review]

**Pull request merged**